### ISSUE TITLE
Properly support S3 structs accross all operators using S3

### ISSUE THREAD

**Issue by sbernauer** [stackabletech/issues#226]
We have defined some common S3 structs in [ADR016](https://docs.stackable.tech/home/contributor/adr/ADR016-definition_of_s3_objects.html).
When implementing support for this in the operators out there we took some shortcuts like ignoring accessStyle and tls settings for now.
We should properly read and implement all S3 attributes out there. Snippet from https://github.com/stackabletech/operator-rs/blob/main/src/commons/s3.rs
```
pub struct S3ConnectionSpec {
    pub host: Option<String>,
    pub port: Option<u16>,
    pub access_style: Option<S3AccessStyle>,
    pub credentials: Option<SecretClassVolume>,
    pub tls: Option<Tls>,
}
```
I had problems using Druid as it didn't support path style access and it needed it for MinIO. I fixed it via https://github.com/stackabletech/druid-operator/pull/245 - just as an implementation idea.
To sum up i think we should support all the attributes, especially
* honor access_style
* honor all tls settings such as
  * if tls should be used
  * ca cert to verify server

in the operators that support S3 i can think of

- [x] Druid
- [x] Trino
- [x] Hive
- [x] Spark-k8s

If a product isn't able to handle the specified settings (e.g. doesn't support path style access) the operator should error out

**Comment by lfrancke** [comment]
@sbernauer Is this done?

**Comment by sbernauer** [comment]
@lfrancke yes it is :) Converted the list in the description to a checklist and checked all items

**Comment by lfrancke** [comment]
Excellent, then.... I'll close it :)
